### PR TITLE
Update benchmarks for JOSS

### LINKED ISSUE #607: Update performance benchmarks plot and discuss results
> The performance plots on the README.md file show that the single precision speed up for CPUs is below 1. This is surprising and should be commented on.
>
> _Originally posted by @funsim in https://github.com/openjournals/joss-reviews/issues/2018#issuecomment-580286479_

### PR DESCRIPTION
**Pull request by ali-ramadhan**
Last minute benchmarks update for JOSS. Resolves #607

The performance benchmarks have been updated from v0.18.0 to v0.34.0.

While it seems that CPU models have gotten faster by about ~32% while large GPU models have slowed down by ~15%, I think that's just because the v0.18.0 benchmarks were run on Supercloud or somewhere with a V100 while the v0.34.0 benchmarks were run on Tartarus where the TITAN V is marginally slower but the CPUs are beefier.

No noticeable regression is pretty great considering how many features and improvements we've added since v0.18.0 and the fact that we haven't really done any profiling and targeted performance optimization.

What seems to have gotten worse is constant overhead costs so small models are slower because of it, but this is something we can tackle when we focus on performance optimization in the future.

benchmark_plots

### CODE REVIEW

**Comment by ali-ramadhan** [comment]
Changed the benchmark scripts slightly but forgot to change the `replace_strings` in `test_benchmarks.jl` so some stuff took way too long and the tests timed out.

This has been a problem in the past leading to test slowdowns if we modify a script but forget to change the `replace_strings` in `test_examples.jl`.

To fix this problem I merged `run_example` and `run_benchmark` into one `run_script` utility that throws an error if a string replacement in `replace_strings` is not found. So we can catch these mistakes before they silently slow down our tests.

So there's some feature creep in this PR...

File: benchmark/benchmark_static_ocean.jl [hunk 1]
```diff
@@ -14,7 +14,7 @@ const timer = TimerOutput()
 Nt = 10  # Number of iterations to use for benchmarking time stepping.
 
 # Model resolutions to benchmarks. Focusing on 3D models for GPU benchmarking.
-            Ns = [(32, 32, 32), (64, 64, 64), (128, 128, 128), (256, 256, 256)]
+            Ns = [(16, 16, 16), (32, 32, 32), (64, 64, 64), (128, 128, 128), (256, 256, 256)]
```
Comment (glwagner) [inline benchmark/benchmark_static_ocean.jl:17]: Is it worth adding `n_threads = [1, 2, 4, 8, 16, 32]` to this benchmark?

Comment (ali-ramadhan) [reply benchmark/benchmark_static_ocean.jl:17]: Yeah I've been meaning to do a multithreading benchmark but couldn't figure out if you can change the number of threads while in Julia. If we can't then we need a bash script to spin up different Julia processes with different numbers of threads.

Either way I think it should be a separate benchmark, e.g. `benchmark_multithreading.jl, but I agree it's a much more useful benchmark.

Comment (glwagner) [reply benchmark/benchmark_static_ocean.jl:17]: Can't we use `ENV["JULIA_NUM_THREADS"] = 2`?

Comment (ali-ramadhan) [reply benchmark/benchmark_static_ocean.jl:17]: Hmmm, not sure if it'll work if you change it on the fly (I thought threads are initialized at Julia startup?) but I'll try.

Comment (glwagner) [reply benchmark/benchmark_static_ocean.jl:17]: I suspect you're right

Comment (glwagner) [reply benchmark/benchmark_static_ocean.jl:17]:
```julia
julia> Threads.nthreads()
8

julia> ENV["JULIA_NUM_THREADS"]
"8"

julia> ENV["JULIA_NUM_THREADS"] = 1
1

julia> Threads.nthreads()
8
```

**Pull request merged**